application/operator/NGSI-LD-operator/alertForSpeedInNGSILD/function.js

```diff
@@ -0,0 +1,77 @@
+var nolmalspeed = 30;
+var averageSpeed1 = 30;
+var averageSpeed2 = 50;
+var highSpeed = 50;
+//
+//  contextEntity: the received entities
+//  publish, query, and subscribe are the callback functions for your own function to interact with the assigned nearby broker
+//      publish:    publish the generated entity, which could be a new entity or the update of an existing entity
+//      query:      query addtional information from the assigned nearby broker
+//      subscribe:  subscribe addtional infromation from the assigned nearby broker
+//
+exports.handler = function(contextEntity, publish, query, subscribe)
+{
+	console.log("enter into the user-defined fog function");	
+	if (contextEntity == null) {
+		return;
+	} 	
+	
+	// ============================== processing ======================================================
+	// processing the received ContextEntity:  
+
+	console.log('ContextEntity.......',contextEntity);
+	
+	var id = contextEntity.id;
+	console.log('id.......',id);	
+	var type = contextEntity.type;
+	console.log('type.......',type);
+	var speed = contextEntity.speed.value;
+	console.log('speed.......',speed);
+	var brand = contextEntity.brandName.value;
+	console.log('brand.......',brand);
+	var resultSpeed = null
+    //PM10
+        if(speed <= 30)
+            resultSpeed =  "speed is slow"
+	else if(speed > averageSpeed1 && speed <= averageSpeed2)
+	    resultSpeed =  "Speed is normal"
+	else 
+	    resultSpeed = "speed is under risk"
+        
+	console.log('Speed......' + resultSpeed);
+	
+	// ============================== publish ======================================================
+        // if you need to publish the generated result, please refer to the following example   
+	/*
+		This example will publish the result is speed is under risk
+	*/ 
+	if(speed > 50)
+	{
+		console.log('publishing started......' );
+		var updateEntity = {};
+		updateEntity.id = id+"ngsild"
```

Comment:
[suggestion] better to take daresult in type and append daresult in entity to createa a new entity in NGSI-LD broker for data analytics result.